safe_transaction_service/history/tests/test_views.py

```diff
@@ -1551,82 +1549,111 @@ def test_safe_collectibles_paginated(self):
-            for result, erc721 in zip(response.data["results"], erc721_list[10:]):
-                self.assertEqual(result["address"], erc721[0])
-                self.assertEqual(int(result["id"]), erc721[1])
+        response = self.client.get(
+            reverse("v2:history:safe-collectibles", args=(safe_address,))
+            + "?limit=5&offset=5",
+            format="json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 11)
+        self.assertEqual(len(response.data["results"]), 5)
+        next = (
+            reverse("v2:history:safe-collectibles", args=(safe_address,))
+            + "?limit=5&offset=10"
+        )
+        previous = (
+            reverse("v2:history:safe-collectibles", args=(safe_address,)) + "?limit=5"
+        )
+        self.assertIn(next, response.data["next"])
+        self.assertIn(previous, response.data["previous"])
+        for result, erc721 in zip(response.data["results"], erc721_list[5:10]):
+            self.assertEqual(result["address"], erc721.address)
+            self.assertEqual(int(result["id"]), erc721.token_id)
+
+        response = self.client.get(
+            reverse("v2:history:safe-collectibles", args=(safe_address,))
+            + "?limit=5&offset=10",
+            format="json",
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data["count"], 11)
+        self.assertEqual(len(response.data["results"]), 1)
+        previous = (
+            reverse("v2:history:safe-collectibles", args=(safe_address,))
+            + "?limit=5&offset=5"
+        )
+        self.assertEqual(response.data["next"], None)
+        self.assertIn(previous, response.data["previous"])
+        for result, erc721 in zip(response.data["results"], erc721_list[10:]):
+            self.assertEqual(result["address"], erc721.address)
+            self.assertEqual(int(result["id"]), erc721.token_id)
+
+        # Check results are sorted
+        # Null address should be first, FF... address should be last
+        erc721_null = ERC721TransferFactory(
+            address=NULL_ADDRESS, token_id=0, to=safe_address
+        )
+        erc721_full = ERC721TransferFactory(
+            address="0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF",
+            token_id=0,
+            to=safe_address,
+        )
+        response = self.client.get(
+            reverse("v2:history:safe-collectibles", args=(safe_address,))
+            + "?limit=1&offset=0",
+            format="json",
+        )
+        self.assertEqual(response.data["results"][0]["address"], erc721_null.address)
+        response = self.client.get(
+            reverse("v2:history:safe-collectibles", args=(safe_address,))
+            + "?limit=1&offset=12",
+            format="json",
+        )
+        self.assertEqual(response.data["results"][0]["address"], erc721_full.address)
```

Comment:
Maybe we must also check if with the same address the response is sorted by token_id.